src/main/redux/sagas/auth.ts

```diff
@@ -780,6 +805,17 @@ function parseRequestFromCustomProtocol(req: Electron.ProtocolRequest)
                     data[key] = value;
                 }
 
+                if (data.state !== getImplicitNonceForImplicitAuthentication()) {
+                    debug("received state parameter is not equal to the nonce sent", "value=", data.state);
+                    debug("the state verification is IGNORED and bypassed to ensure a legacy compatibility with all OPDS OAUTH2 server");
+                    // TODO: improve this and enable the state verification
```

Comment:
I think a possibility for enabling backwards-compatible behaviour would be to only ignore the `state` nonce if it is missing in the response from the OAuth server. Something like:

```js
// https://auth0.com/docs/secure/attack-protection/state-parameters
if (data.state) {
    if (data.state === getImplicitNonceForImplicitAuthentication()) {
        debug("State parameter was included in callback response and is VERIFIED and CORRECT");
    }
    else {
        // Todo: Reject the response, user is potentially compromised
        debug("received state parameter is not equal to the nonce sent", "value=", data.state);
    }
}
else {
    debug("State verification missing in response - verification checks IGNORED and bypassed to ensure legacy compatibility with all OPDS OAUTH2 server");
}
```

Reply:
Thinking about it a little more, this probably doesn't provide much additional security. If a user's connection is compromised, a malicious actor could just omit the `state` parameter in the crafted response and the check would be bypassed.